packages/vitest/src/node/watch-filter.ts

```diff
@@ -0,0 +1,160 @@
+import readline from 'node:readline'
+import c from 'picocolors'
+import { stdout } from '../utils'
+
+const MAX_RESULT_COUNT = 10
+const SELECTION_MAX_INDEX = 7
+const ESC = '\u001B['
+
+type FilterFunc = (keyword: string) => Promise<string[]> | string[]
+
+export class WatchFilter {
+  private filterRL: readline.Interface
+  private currentKeyword: string | undefined = undefined
+  private message: string
+  private results: string[] = []
+  private selectionIndex = -1
+  private onKeyPress?: (str: string, key: any) => void
+
+  constructor(message: string) {
+    this.message = message
+    this.filterRL = readline.createInterface({ input: process.stdin, escapeCodeTimeout: 50 })
+    readline.emitKeypressEvents(process.stdin, this.filterRL)
+    if (process.stdin.isTTY)
+      process.stdin.setRawMode(true)
+  }
+
+  public async filter(filterFunc: FilterFunc): Promise<string | undefined> {
+    stdout().write(`${c.cyan('?')} ${c.bold(this.message)} › `)
+    return new Promise((resolve) => {
+      this.onKeyPress = this.filterHandler(filterFunc, (result) => {
+        this.close()
+        resolve(result)
+      })
+      process.stdin.on('keypress', this.onKeyPress)
+    })
+  }
+
+  private filterHandler(filterFunc: FilterFunc, onSubmit: (result?: string) => void) {
+    return async (str: string, key: any) => {
+      switch (true) {
+        case key.sequence === '\x7F':
+          if (this.currentKeyword && this.currentKeyword?.length > 1)
+            this.currentKeyword = this.currentKeyword?.slice(0, -1)
+
+          else
+            this.currentKeyword = undefined
+
+          break
+        case key?.ctrl && key?.name === 'c':
+        case key?.name === 'escape':
+          this.cancel()
+          onSubmit(undefined)
+          break
```

Comment:
It looks like Ctrl-C behavior between `inputNamePattern` case and `inputFilePattern` case are different?
For `inputFilePattern`, it reruns everything (which personally looks odd but maybe okay since it's same as current main).
For `inputNamePattern`, I'm not sure what's happening actually.

Can someone double check the behavior of cancel/escape is expected?

Reply:
@hi-ogawa 
Thank you for catching this. 
This was a bug and I've fixed it in 43da5fe.